### ISSUE TITLE
Unable to Confirm an Accounts KYC Granted Status before sending Tokens

### ISSUE THREAD

**Issue by bugbytesinc** [hashgraph/hedera-services#579]
**Summary of the defect**

If you create an account and query the account info to discover the KYC status of the account, you will get no entry for the token.  This is expected.  However after you "Grant" kyc to the target account, you still have no Token Record returned from subsequent info queries until AFTER you send tokens to the account.  Note: you can't send tokens if KYC is not granted.  

In other words, the info query returns false negatives with regards to KYC status.

**How to reproduce**

- Create a token
- Create an account with KYC turned on
- query info the account, no entries for the token
- (at this point you could try to transfer some tokens to the account, it will fail)
- grant KYC access for the token on the account
- query info the account, still no entries for the token
- transfer some tokens to the Account
- query info for the account, it will now have an entry for granted KYC for the token (as well as the balance)

**Test Log Exertp**
```

// Get Info: does not show Token KYC Status

9/18/2020 1:49:17 PM  QX ASK → { "cryptoGetInfo": { "header": { "payment": { "bodyBytes": "" }, "responseType": "COST_ANSWER" }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:17 PM  RX:(00)  { "cryptoGetInfo": { "header": { "responseType": "COST_ANSWER", "cost": "8" } } }
9/18/2020 1:49:17 PM  QX PYMT  { "transactionID": { "transactionValidStart": { "seconds": "1600436955", "nanos": 873379400 }, "accountID": { "accountNum": "1001" } }, "nodeAccountID": { "accountNum": "3" }, "transactionFee": "2900000000", "transactionValidDuration": { "seconds": "120" }, "cryptoTransfer": { "transfers": { "accountAmounts": [ { "accountID": { "accountNum": "1001" }, "amount": "-8" }, { "accountID": { "accountNum": "3" }, "amount": "8" } ] } } }
9/18/2020 1:49:17 PM  └─ QRY → { "cryptoGetInfo": { "header": { "payment": { "sigMap": { "sigPair": [ { "pubKeyPrefix": "CqjiEGTG", "ed25519": "rPdHwZ3ZQ3AgoqPskGNPFs+TT0yOU92oO6DH0NbaWbUv7S5ApjzylvfsBxLUHosdKyO1JZ4n435pqk1Rt87qCA==" } ] }, "bodyBytes": "ChMKDAjb9ZL7BRDI7LqgAxIDGOkHEgIYAxiA+unmCiICCHhyEwoRCgcKAxjpBxAPCgYKAhgDEBA=" } }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:17 PM  RX:(00)  { "cryptoGetInfo": { "header": { }, "accountInfo": { "accountID": { "accountNum": "3429" }, "contractAccountID": "0000000000000000000000000000000000000d65", "proxyAccountID": { }, "key": { "ed25519": "BrWOE/Z4JhGXqeB+CNeEfn5nu0NkTswmA1g7SBtTJN0=" }, "balance": "14", "generateSendRecordThreshold": "9223372036854775807", "generateReceiveRecordThreshold": "9223372036854775807", "expirationTime": { "seconds": "1608326955" }, "autoRenewPeriod": { "seconds": "7890000" } } } }

// Grant KYC to Account

9/18/2020 1:49:17 PM  TX BODY  { "transactionID": { "transactionValidStart": { "seconds": "1600436955", "nanos": 890825400 }, "accountID": { "accountNum": "1001" } }, "nodeAccountID": { "accountNum": "3" }, "transactionFee": "2900000000", "transactionValidDuration": { "seconds": "120" }, "tokenGrantKyc": { "token": { "tokenId": { "tokenNum": "3432" } }, "account": { "accountNum": "3429" } } }
9/18/2020 1:49:17 PM  └─ SIG → { "sigMap": { "sigPair": [ { "pubKeyPrefix": "CqjiEGTG", "ed25519": "PyazAtZIwI6IQiHJPxJALDljZ0irXBs+hAv8krytiAwHbB8D+qIc7cqAxMaUvNO9MeNzhSlBNosL7zm4n/POAA==" }, { "pubKeyPrefix": "6sy7fq3J", "ed25519": "+BovPNiIWiRoA6bhY11JWlpcIcHmuDoAcmkHGjXJeJS/zOoMULayvAFxZzmZ53NVQpTv9KXlscutOir5IquaCQ==" } ] }, "bodyBytes": "ChMKDAjb9ZL7BRC41eOoAxIDGOkHEgIYAxiA+unmCiICCHiKAgwKBQoDGOgaEgMY5Ro=" }
9/18/2020 1:49:17 PM  RX:(00)  { }
9/18/2020 1:49:17 PM  TX     → { "transactionGetReceipt": { "transactionID": { "transactionValidStart": { "seconds": "1600436955", "nanos": 890825400 }, "accountID": { "accountNum": "1001" } } } }
9/18/2020 1:49:17 PM  RX:(00)  { "transactionGetReceipt": { "header": { }, "receipt": { "status": "UNKNOWN" } } }
9/18/2020 1:49:17 PM  RX:(01)  { "transactionGetReceipt": { "header": { }, "receipt": { "status": "SUCCESS", "exchangeRate": { "currentRate": { "hbarEquiv": 1, "centEquiv": 12, "expirationTime": { "seconds": "4102444800" } }, "nextRate": { "hbarEquiv": 1, "centEquiv": 15, "expirationTime": { "seconds": "4102444800" } } } } } }

// Query Info of Account, Note: no KYC Status: Is the KYC status Granted or Not?

9/18/2020 1:49:17 PM  QX ASK → { "cryptoGetInfo": { "header": { "payment": { "bodyBytes": "" }, "responseType": "COST_ANSWER" }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:17 PM  RX:(00)  { "cryptoGetInfo": { "header": { "responseType": "COST_ANSWER", "cost": "8" } } }
9/18/2020 1:49:17 PM  QX PYMT  { "transactionID": { "transactionValidStart": { "seconds": "1600436956", "nanos": 125145500 }, "accountID": { "accountNum": "1001" } }, "nodeAccountID": { "accountNum": "3" }, "transactionFee": "2900000000", "transactionValidDuration": { "seconds": "120" }, "cryptoTransfer": { "transfers": { "accountAmounts": [ { "accountID": { "accountNum": "1001" }, "amount": "-8" }, { "accountID": { "accountNum": "3" }, "amount": "8" } ] } } }
9/18/2020 1:49:17 PM  └─ QRY → { "cryptoGetInfo": { "header": { "payment": { "sigMap": { "sigPair": [ { "pubKeyPrefix": "CqjiEGTG", "ed25519": "YaHaGx2N67R05kr+cVElYTBQfDNjc3+uihe/BaLDGSXGr2EnTHPvyjFdmXaYoGWuGIL6+qKq3yGHT2I/Kk7vDQ==" } ] }, "bodyBytes": "ChIKCwjc9ZL7BRCco9Y7EgMY6QcSAhgDGID66eYKIgIIeHITChEKBwoDGOkHEA8KBgoCGAMQEA==" } }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:17 PM  RX:(00)  { "cryptoGetInfo": { "header": { }, "accountInfo": { "accountID": { "accountNum": "3429" }, "contractAccountID": "0000000000000000000000000000000000000d65", "proxyAccountID": { }, "key": { "ed25519": "BrWOE/Z4JhGXqeB+CNeEfn5nu0NkTswmA1g7SBtTJN0=" }, "balance": "14", "generateSendRecordThreshold": "9223372036854775807", "generateReceiveRecordThreshold": "9223372036854775807", "expirationTime": { "seconds": "1608326955" }, "autoRenewPeriod": { "seconds": "7890000" } } } }

// Send Tokens to Acocunt

9/18/2020 1:49:17 PM  TX BODY  { "transactionID": { "transactionValidStart": { "seconds": "1600436956", "nanos": 136330900 }, "accountID": { "accountNum": "1001" } }, "nodeAccountID": { "accountNum": "3" }, "transactionFee": "2900000000", "transactionValidDuration": { "seconds": "120" }, "tokenTransfers": { "tokenTransfers": [ { "token": { "tokenId": { "tokenNum": "3432" } }, "transfers": [ { "accountID": { "accountNum": "3430" }, "amount": "-40000000000" }, { "accountID": { "accountNum": "3429" }, "amount": "40000000000" } ] } ] } }
9/18/2020 1:49:17 PM  └─ SIG → { "sigMap": { "sigPair": [ { "pubKeyPrefix": "CqjiEGTG", "ed25519": "LQmPZGTrhutAoYoYb9CKMUmk0BhmQruGvXl3m43g4aAgzOH2WtcnrIYVDXeRmqj0FekDDTj7ovVEGWOU4fKPBg==" }, { "pubKeyPrefix": "zatT9v00", "ed25519": "34W3Td+trytEbNAU+U7azDXp7SgT1/7TFIvYjuHrhDEdspkeHQdwCULvQR8srMHfBf4zBhF/4NMiOKikRIKzCQ==" } ] }, "bodyBytes": "ChIKCwjc9ZL7BRCU/YBBEgMY6QcSAhgDGID66eYKIgIIePIBJQojCgUKAxjoGhIMCgMY5hoQ/7/8gqoCEgwKAxjlGhCAwPyCqgI=" }
9/18/2020 1:49:17 PM  RX:(00)  { }
9/18/2020 1:49:17 PM  TX     → { "transactionGetReceipt": { "transactionID": { "transactionValidStart": { "seconds": "1600436956", "nanos": 136330900 }, "accountID": { "accountNum": "1001" } } } }
9/18/2020 1:49:17 PM  RX:(00)  { "transactionGetReceipt": { "header": { }, "receipt": { "status": "UNKNOWN" } } }
9/18/2020 1:49:18 PM  RX:(01)  { "transactionGetReceipt": { "header": { }, "receipt": { "status": "SUCCESS", "exchangeRate": { "currentRate": { "hbarEquiv": 1, "centEquiv": 12, "expirationTime": { "seconds": "4102444800" } }, "nextRate": { "hbarEquiv": 1, "centEquiv": 15, "expirationTime": { "seconds": "4102444800" } } } } } }

// Query Info of Account: Now KYC Status is Visible

9/18/2020 1:49:18 PM  QX ASK → { "cryptoGetInfo": { "header": { "payment": { "bodyBytes": "" }, "responseType": "COST_ANSWER" }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:18 PM  RX:(00)  { "cryptoGetInfo": { "header": { "responseType": "COST_ANSWER", "cost": "8" } } }
9/18/2020 1:49:18 PM  QX PYMT  { "transactionID": { "transactionValidStart": { "seconds": "1600436956", "nanos": 362058600 }, "accountID": { "accountNum": "1001" } }, "nodeAccountID": { "accountNum": "3" }, "transactionFee": "2900000000", "transactionValidDuration": { "seconds": "120" }, "cryptoTransfer": { "transfers": { "accountAmounts": [ { "accountID": { "accountNum": "1001" }, "amount": "-8" }, { "accountID": { "accountNum": "3" }, "amount": "8" } ] } } }
9/18/2020 1:49:18 PM  └─ QRY → { "cryptoGetInfo": { "header": { "payment": { "sigMap": { "sigPair": [ { "pubKeyPrefix": "CqjiEGTG", "ed25519": "147VCltIkURenZqrWpFpCd9hF2PJp/JUAXHMJX1KnS+xfsSC0d4QsNd/5cd/TtrLExthJ+EiZ9ezmdOJhL33Bg==" } ] }, "bodyBytes": "ChMKDAjc9ZL7BRDoptKsARIDGOkHEgIYAxiA+unmCiICCHhyEwoRCgcKAxjpBxAPCgYKAhgDEBA=" } }, "accountID": { "accountNum": "3429" } } }
9/18/2020 1:49:18 PM  RX:(00)  { "cryptoGetInfo": { "header": { }, "accountInfo": { "accountID": { "accountNum": "3429" }, "contractAccountID": "0000000000000000000000000000000000000d65", "proxyAccountID": { }, "key": { "ed25519": "BrWOE/Z4JhGXqeB+CNeEfn5nu0NkTswmA1g7SBtTJN0=" }, "balance": "14", "generateSendRecordThreshold": "9223372036854775807", "generateReceiveRecordThreshold": "9223372036854775807", "expirationTime": { "seconds": "1608326955" }, "autoRenewPeriod": { "seconds": "7890000" }, "tokenRelationships": [ { "tokenId": { "tokenNum": "3432" }, "balance": "40000000000", "symbol": "RMOYCTNQUFLIOBNYXPMM", "kycStatus": "Granted", "freezeStatus": "Unfrozen" } ] } } }
...
```
**Environment:**
- WSL2 Docker Ubuntu 20.04
- Hedera Services Version: TAG=v0.8.0-rc1-112-g22b6f40f-dirty

**Comment by bugbytesinc** [comment]
It appears that  `v0.8.0-rc1-138-gc357a75b-dirty` has fixed this issue ** however** a similar problem still exists with freeze/unfreeze implementation.

**Comment by Daniel-K-Ivanov** [comment]
Will be fixed in #613
